### DISCUSSION TITLE
Iron ore fines 62% Fe CFR Futures

### DISCUSSION THREAD

**Discussion by maykon17** [alvarobartt/investpy#415] [Q&A]
Hi! I'm currently extracting historical market data and investpy is very useful to me, but the commodity "Iron ore fines 62% Fe CFR Futures" doesn't seem to be available in investpy, I didn't find the historical data and the smybol in commodities list. Could anyone help me? Thanks. @alvarobartt

**Comment by knickr** [comment]
Its there if you search for Iron Ore in investpy

{"id_": 961729, "name": "Iron ore fines 62% Fe CFR Futures", "symbol": "TIOc1", "country": null, "tag": "/commodities/iron-ore-62-cfr-futures", "pair_type": "commodities", "exchange": "CME"}

**Reply by jefferygao1984** [reply]
#No luck.

#first attempt
search_result = investpy.search_quotes(text='iron ore', products=['commodities'])
print(search_result)

[<investpy.utils.search_obj.SearchObj object at 0x7f15eeeb3ed0>, <investpy.utils.search_obj.SearchObj object at 0x7f15eeeb3f10>, <investpy.utils.search_obj.SearchObj object at 0x7f15eeeb3f50>]

#second attempt
now = datetime.datetime.now()
from_date = '01/01/1900'
to_date = str(now.day)+"/"+str(now.month)+"/"+str(now.year)
commodity = 'Iron ore fines 62% Fe CFR Futures'

investpy.commodities.get_commodity_historical_data(commodity, from_date, to_date, country=None, as_json=False, order='ascending', interval='Daily')

...i wish investpy allow extraction via ISIN or symbol. using 'name' is fraught with ambiguity.

**Comment by knickr** [comment]
In your first attempt you need to limit the results to 1 as there are two futures contracts one in Singapore and one in the US (CME).

search_result = investpy.search_quotes(text='Iron Ore', products=['commodities'], countries=['United States'], **n_results=1**

then you can use historical_data = search_result.retrieve_historical_data(from_date='DD/MM/YY', to_date='DD/MM/YY')

Seems that Iron Ore is not in the list of commodities that Investpy uses to check if its valid even though it appears on Investing.com under the 'Metals' group of commodities which is why trying to get it without using the search fails.  Agree that it should use a symbol or Investing.com ID in addition to 'name' to get historical data.

**Reply by jefferygao1984** [reply]
Kewl! Thanks for the workaround!

And yeah, I got stuck when get_commodities didn't return iron ore.

Reason I omitted country and no_results was because I know there are three futures contracts for 62% iron ore fines:
1. SGXIOSc1, Singapore (USD), from 30 APR 2009
2. TIOc1, United States (USD), from 25 OCT 2010
3. DCIOF2, China (CNY), from 23 OCT 2013

It clearly found all three but returned the hexidecimal index for the search object instead. 

And SGXIOSc1 is screwy in that it's classified under group 'Energy' by investing.com.
